invenio_records_resources/records/api.py

```diff
@@ -177,7 +178,9 @@ def dumps(self):
     @property
     def ext(self):
         """File extension."""
-        ext = mimetypes.guess_extension(self.object_model.mimetype)
```

Comment:
```python
# The ``ext`` property is used to in search to aggregate file types and we want e.g. both ``.jpeg`` and ``.jpg`` to be aggregated under ``.jpg``
ext = mimetypes.guess_extension(self.object_model.mimetype)
if ext is not None:
    return ext[1:]

# Support non-standard file extensions that cannot be guessed
_, ext = os.path.splitext(self.key)
if ext and len(ext) <= 5:
    return ext.lower()
```

Reply:
I see, thanks for the clarification.
However to make this work we would also need to change [this](https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/utils.py#L57) to not return "application/octet-stream". I will do the change

Reply:
Decided not to remove the fallback mimetype as it might cause issues on files serving, instead we check if the mimetype is set to the fallback to try to get the extenstion from the filename or not.